clouddriver-kubernetes/src/main/groovy/com/netflix/spinnaker/clouddriver/kubernetes/api/KubernetesClientApiAdapter.groovy

```diff
@@ -0,0 +1,142 @@
+/*
+ * Copyright 2016 Google, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License")
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.netflix.spinnaker.clouddriver.kubernetes.api
+
+import com.netflix.spectator.api.Clock
+import com.netflix.spectator.api.Registry
+import com.netflix.spinnaker.clouddriver.kubernetes.deploy.KubernetesUtil
+import com.netflix.spinnaker.clouddriver.kubernetes.deploy.exception.KubernetesOperationException
+import com.netflix.spinnaker.clouddriver.kubernetes.security.KubernetesApiClientConfig
+import groovy.util.logging.Slf4j
+import io.kubernetes.client.ApiClient
+import io.kubernetes.client.ApiException
+import io.kubernetes.client.Configuration
+import io.kubernetes.client.apis.AppsV1beta1Api
+import io.kubernetes.client.models.*
+
+import java.util.concurrent.TimeUnit
+
+class KubernetesClientApiAdapter {
+  String account
+
+  static final int RETRY_COUNT = 20
+  static final long RETRY_MAX_WAIT_MILLIS = TimeUnit.SECONDS.toMillis(10)
+  static final long RETRY_INITIAL_WAIT_MILLIS = 100
+  static final int API_CALL_TIMEOUT_SECONDS = 60
+  static final String DEPLOYMENT_ANNOTATION = "deployment.kubernetes.io"
+  final Registry spectatorRegistry
+  final Clock spectatorClock
+  final ApiClient client
+  final AppsV1beta1Api apiInstance
+
+  public spectatorRegistry() { return spectatorRegistry }
+
+  KubernetesClientApiAdapter(String account, KubernetesApiClientConfig config, Registry spectatorRegistry) {
+    if (!config) {
+      throw new IllegalArgumentException("Config may not be null.")
+    }
+
+    this.account = account
+    this.spectatorRegistry = spectatorRegistry
+    this.spectatorClock = spectatorRegistry.clock()
+
+    client = config.getApiCient()
+    Configuration.setDefaultApiClient(client)
+    apiInstance = new AppsV1beta1Api();
+  }
+
+  KubernetesOperationException formatException(String operation, String namespace, ApiException e) {
+    account ? new KubernetesOperationException(account, "$operation in $namespace", e) :
+      new KubernetesOperationException("$operation in $namespace", e)
+  }
+
+  KubernetesOperationException formatException(String operation, ApiException e) {
+    account ? new KubernetesOperationException(account, "$operation", e) :
+      new KubernetesOperationException("$operation", e)
+  }
+
+  Boolean blockUntilResourceConsistent(Object desired, Closure<Long> getGeneration, Closure getResource) {
```

Comment:
Are these copied directly from the other Adaptor? If so, they should probably be pulled into a separate class and be shared.